upgrade_guide/migrating_from_astyanax/language_level_changes/README.md

```diff
@@ -0,0 +1,75 @@
+# Language change : from Thrift to CQL
+The data model changes when using *CQL* (Cassandra Query Language).
+*CQL* is providing an abstraction of the low-level data stored in *Cassandra*, in
+opposition to *Thrift* that aims to expose the low-level data structure directly.
+[But note that this changes with Cassandra 3’s new storage engine.](http://www.datastax.com/2015/12/storage-engine-30)
+
+*Thrift* exposes *Keyspaces*, and these *Keyspaces* contain *Column Families*. A
+*ColumnFamily* contains *Rows* in which each *Row* has a list of an arbitrary number
+of column-values. With *CQL*, the data is **tabular**, *ColumnFamily* gets viewed
+as a *Table*, the **Table Rows** get a **fixed and finite number of named columns**.
+*Thrift*’s columns inside the *Rows* get distributed in a tabular way through the
+_Table Rows_. See the following figure :
+
+```ditaa
+                                                     Thrift
+             /-                                                                                          -\
+             |                                                                                            |
+             |  /------------\              /---------------+---------------+---------------+---------+   |
+             |  |    cRED    |              |cFA0 Col : 1   |     Col : 2   |     Col : 3   |         |   |
+             |  | Key : 1    | ---------->  +---------------+---------------+---------------+ ...     |   +--> One Thrift
+             |  |            |              |c1AB Val : 'a' |     Val : 'b' |     Val : 'c' |         |   |    ROW
+             |  \------------/              \---------------+---------------+---------------+---------+   |
+             |                                                                                            |
+One Thrift   |                                                                                           -/
+COLUMNFAMILY |          
+             |  
+             |  /------------\              /---------------+---------------+---------+
+             |  |            |              |     Col : 1   |     Col : 2   |         |
+             |  | Key : 2    | ---------->  +---------------+---------------+ ...     |
+             |  |            |              |     Val : 'a' |     Val : 'b' |         |
+             |  \------------/              \---------------+---------------+---------+
+             |  
+             \- 
+
+
+                      -----------------------------------------------------------------------
+
+
+                                                     CQL
+                                                     
+             /-                              
+             |                               
+             |  /--------------------+---------------------------------+-----------------------------\
```

Comment:
This is not the way I would intuitively represent a CQL row. It would be more like this:

<table>
<tr><th>key</th><th>column: 1</th><th>column: 2</th><th>column: 3</th></tr>
<tr><td>cRED   1</td><td>'a'</td><td>'b'</td><td>'c'</td></tr>
<tr><td>cRED   2</td><td>'a'</td><td>'b'</td><td>null</td></tr>
</table>